react/features/video-quality/middleware.js

```diff
@@ -46,21 +47,42 @@ StateListenerRegistry.register(
         const { reducedUI } = state['features/base/responsive-ui'];
         const _shouldDisplayTileView = shouldDisplayTileView(state);
         const thumbnailSize = state['features/filmstrip']?.tileViewDimensions?.thumbnailSize;
+        const participantCount = getParticipantCount(state);
 
         return {
             displayTileView: _shouldDisplayTileView,
+            participantCount,
             reducedUI,
             thumbnailHeight: thumbnailSize?.height
         };
     },
-    /* listener */ ({ displayTileView, reducedUI, thumbnailHeight }, { dispatch, getState }) => {
-        const { maxReceiverVideoQuality } = getState()['features/base/conference'];
+    /* listener */ ({ displayTileView, participantCount, reducedUI, thumbnailHeight }, { dispatch, getState }) => {
+        const state = getState();
+        const { maxReceiverVideoQuality } = state['features/base/conference'];
+        const { maxFullResolutionParticipants = 2 } = state['features/base/config'];
+
         let newMaxRecvVideoQuality = VIDEO_QUALITY_LEVELS.HIGH;
 
         if (reducedUI) {
             newMaxRecvVideoQuality = VIDEO_QUALITY_LEVELS.LOW;
         } else if (displayTileView && !Number.isNaN(thumbnailHeight)) {
             newMaxRecvVideoQuality = getNearestReceiverVideoQualityLevel(thumbnailHeight);
+
+            // Override HD level calculated for the thumbnail height when # of participants threshold is exceeded
+            if (maxReceiverVideoQuality !== newMaxRecvVideoQuality && maxFullResolutionParticipants !== -1) {
```

Comment:
Shouldn't this check if `newMaxRecvVideoQuality` is > than `newMaxRecvVideoQuality` ? Maybe I'm reading this wrong but it looks like if the current setting is standard and a new participant joins (or the window is resized) which makes the tails just small enough to go to low, we'd force it back up to standard.

Reply:
There is such check at line 75. This if is there only to limit the amount of log messages printed on each change.

Reply:
No, it's fine, I just missed it, sorry!